editoast/src/views/rolling_stocks.rs

```diff
@@ -382,20 +429,69 @@ pub mod tests {
 
         //Check rolling_stock creation
         assert_eq!(post_response.status(), StatusCode::OK);
-        let rolling_stock: RollingStock = read_body_json(post_response).await;
-        assert_eq!(rolling_stock.name, "fast_rolling_stock");
+        let rolling_stock_full: RollingStockFull = read_body_json(post_response).await;
+        let rolling_stock_id = rolling_stock_full.rolling_stock.id;
+        assert_eq!(rolling_stock_full.rolling_stock.name, "fast_rolling_stock");
 
         //Check rolling_stock deletion
-        let delete_request = rolling_stock_delete_request(rolling_stock.id);
+        let delete_request = rolling_stock_delete_request(rolling_stock_id);
         let delete_response = call_service(&app, delete_request).await;
         assert_eq!(delete_response.status(), StatusCode::NO_CONTENT);
 
-        //Check object does not exist anymore
-        let get_request = rolling_stock_get_request(0);
+        //Check rolling_stock does not exist anymore
+        let get_request = rolling_stock_get_request(rolling_stock_id);
         let get_response = call_service(&app, get_request).await;
         assert_eq!(get_response.status(), StatusCode::NOT_FOUND);
     }
 
+    #[rstest]
+    async fn update_and_delete_locked_rolling_stock_fails(
+        db_pool: Data<Pool<ConnectionManager<diesel::PgConnection>>>,
+    ) {
+        let app = create_test_service().await;
+        let rolling_stock: RollingStockModel = get_fast_rolling_stock();
+        let post_response = call_service(
+            &app,
+            TestRequest::post()
+                .uri("/rolling_stock?locked=true")
+                .set_json(rolling_stock)
+                .to_request(),
+        )
+        .await;
+        let rolling_stock_full: RollingStockFull = read_body_json(post_response).await;
+        let rolling_stock_id = rolling_stock_full.rolling_stock.id;
+
+        //Check rolling_stock update fails
+        let patch_response = call_service(
+            &app,
+            TestRequest::patch()
+                .uri(format!("/rolling_stock/{}", &rolling_stock_full.rolling_stock.id).as_str())
+                .set_json(&rolling_stock_full)
+                .to_request(),
+        )
+        .await;
+        assert_eq!(patch_response.status(), StatusCode::BAD_REQUEST);
+        let patch_body: JsonValue = read_body_json(patch_response).await;
+        let is_locked_error: InternalError =
+            RollingStockError::RollingStockIsLocked { rolling_stock_id }.into();
+        assert_eq!(to_value(&is_locked_error).unwrap(), patch_body);
```

Comment:
There is a couple macro defs that might make such assertions more concise and avoid `JsonValue` here:

https://github.com/DGEXSolutions/osrd/blob/c8a1770ca54769c13e51c60931707d364e1ec781/editoast/src/views/mod.rs#L93-L126

and find their usage here:

https://github.com/DGEXSolutions/osrd/blob/c8a1770ca54769c13e51c60931707d364e1ec781/editoast/src/views/pathfinding/mod.rs#L519

Reply:
Thanks for that, makes testing errors a whole lot easier.